packages/salesforcedx-apex-debugger/src/adapter/apexDebug.ts

```diff
@@ -121,15 +121,22 @@ export enum ApexVariableKind {
 export class ApexVariable extends Variable {
   public readonly declaredTypeRef: string;
   public readonly type: string;
+  public readonly indexedVariables?: number;
   private readonly slot: number;
   private readonly kind: ApexVariableKind;
 
   constructor(
     value: Value,
     kind: ApexVariableKind,
-    variableReference?: number
+    variableReference?: number,
+    numOfChildren?: number
   ) {
-    super(value.name, ApexVariable.valueAsString(value), variableReference);
+    super(
+      value.name,
+      ApexVariable.valueAsString(value),
+      variableReference,
+      numOfChildren
```

Comment:
This ApexVariable class extends VS Code debug protocol's Variable which has these fields:

![debugprotocolvariable](https://user-images.githubusercontent.com/7286437/32526403-f6c65256-c3dd-11e7-9e6c-2f9c1c3de747.png)

If VS Code receives a large indexedVariables number, it will automatically paginate with 100 in each page. So, the important thing for our debugger adapter is to save the collection's reference size and assign that to indexedVariables property, which is what we're doing here.